### ISSUE TITLE
Why were the {M, Un}arshalJSON Methods for Enums Removed in Proto3

### ISSUE THREAD

**Issue by frankgreco** [gogo/protobuf#515]
Is appears like like the `MarshalJSON` and `UnmarshalJSON` methods for enums were removed in proto3 as evidence by [this](https://github.com/gogo/protobuf/blob/e5d5b02ea0b3c4fd2a996e909ffa0f3421d6c91b/protoc-gen-gogo/generator/generator.go#L1563-L1583) code. 

What is the workaround for proto3? Currently I have to manually implement these methods.

**Comment by frankgreco** [comment]
@donaldgraham @jmarais Any update?

**Comment by donaldgraham** [comment]
Could you ask golang/protobuf?  As far as @awalterschulze can remember, we simply merged their code.

**Comment by frankgreco** [comment]
@donaldgraham I already asked there and they directed me here. https://github.com/golang/protobuf/issues/750

**Comment by frankgreco** [comment]
@awalterschulze @donaldgraham have you looked at the code that I linked in the issue because it doesn't seem like `golang/protobuf` code.

**Comment by awalterschulze** [comment]
https://github.com/golang/protobuf/blob/master/protoc-gen-go/generator/generator.go#L1422-L1432

**Comment by awalterschulze** [comment]
No, because I am incredibly lazy :) It is one of my best qualities.
I was hoping that pushing this issue to golang/protobuf would make it go away, because typically we didn't try to change anything with proto3 and were simply merging in changes.  We get these types of issues a lot.  Sorry for not digging deep enough.

In the link to golang protobuf you will see that golang/protobuf also checks for proto3 and then does not generate the UnmarshalJSON method.
https://github.com/golang/protobuf/blob/master/protoc-gen-go/generator/generator.go#L1422-L1432
I can't remember when they removed or if they ever had the MarshalJSON method, but there I was trying to be symmetrical with my merge.

I hope that makes sense.
